pkg/api/v1alpha1/nutanixdatacenterconfig_types.go

```diff
@@ -0,0 +1,149 @@
+// Important: Run "make generate" to regenerate code after modifying this file
+// json tags are required; new fields must have json tags for the fields to be serialized
+
+package v1alpha1
+
+import (
+	"crypto/x509"
+	"errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NutanixDatacenterConfigSpec defines the desired state of NutanixDatacenterConfig
+type NutanixDatacenterConfigSpec struct {
+	// Endpoint is the Endpoint of Nutanix Prism Central
+	// +kubebuilder:validation:Required
+	Endpoint string `json:"endpoint"`
+
+	// Port is the Port of Nutanix Prism Central
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=9440
+	Port int `json:"port"`
+
+	// User is the User name for Nutanix Prism Central
+	// +kubebuilder:validation:Required
+	User string `json:"user"`
+
+	// Password is the Password for Nutanix Prism Central
+	// +kubebuilder:validation:Required
+	Password string `json:"password"`
```

Comment:
Can we read username and password from environment variables and store them in a Kubernetes secret?
Ideally, we shouldn't expose credentials in the cluster config

Reply:
We indeed plan on using secret but were using the config as an intermediary object; this is not really necessary so we can move it out.